gfx/common/x11_common.c

```diff
@@ -306,9 +306,17 @@ static void x11_suspend_screensaver_xdg_screensaver(Window wnd, bool enable)
 
 void x11_suspend_screensaver(Window wnd, bool enable)
 {
-    x11_suspend_screensaver_xdg_screensaver(wnd, enable);
 #ifdef HAVE_DBUS
     x11_suspend_screensaver_dbus(enable);
+    if (dbus_screensaver_cookie == 0)
+    {
+        // Only attempt xdg-screensaver method if dbus didn't succeed
+#endif
+
+        x11_suspend_screensaver_xdg_screensaver(wnd, enable);
+
+#ifdef HAVE_DBUS
+    }
```

Comment:
No braces is indeed more likely to error, but if that brace requires another ifdef, then there's a lot of added noise.

Proposed compromise: Put the opening brace outside the ifdef. You want the braces, I don't want the ifdef; that combination makes us both happy.

(I think the local coding style is no braces unless needed, but I'm not sure, so I'll accept anything here.)


Yes, uninhibiting is currently broken, but that bug is currently only in suspend-dbus(). Let's keep it that way, so it's easy to fix properly later.

My prefered solution would be adding a return value to suspend-dbus() telling whether it worked, but if you prefer something else (for example comparing the result of the equality operator), then I'm fine with that too.